### PR TITLE
Make it possible to run the wpt ref tests from this repo

### PR DESCRIPTION
**Pull request by greggman**
See included docs

Note: I'm fine if you reject this PR as not part of the offcial CTS repo but I thought I'd post it if others want to use it.

I found it slow going to run the wpt ref tests by having to generate them into another repo after every single edit. I thought maybe I could whip something up that let me run them locally. This is the result. It's working for me. I used [playwrite-core](https://www.npmjs.com/package/playwright-core) which claims to be cross browser. I didn't check if I could actually run WebKit or Firefox but they should be easy to add.

I started on to implement things like how `class="reftest-wait"` but there's some things I don't get. Maybe bugs in our cts test. According to the wpt `class="reftest-wait"` does the following

> * Wait for the load event to fire and fonts to load.
> * Wait for pending paints to complete.
> * Fire an event named TestRendered at the root element, with the bubbles attribute set to true.
> * Wait for the reftest-wait class to be removed from the root element.
> * Wait for pending paints to complete.
> * Screenshot the viewport.

But right now (maybe my bug) `ref/canvas_image_rendering-ref.html` has `class="reftest-wait"` but here's no code in there to remove the `reftest-wait` from the root element. I guess we need to add that.

Issue: None

<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by austinEng** [comment]
I skimmed the code, but IMO it's great if we can run the reftests even if the comparisons aren't exactly how they work in WPT. Should be very helpful for local testing.

Kai is probably a better reviewer here

**Comment by kainino0x** [comment]
> But right now (maybe my bug) `ref/canvas_image_rendering-ref.html` has `class="reftest-wait"` but here's no code in there to remove the `reftest-wait` from the root element. I guess we need to add that.

FTR, fixed (incorrectly) in #2140 and then (correctly) in 0cb55ff527a95c69251abed5390d0a26c42a7ab9.

**Review by kainino0x — APPROVED** [review]
This is awesome, short and sweet! Playwright looks quite nice. Just one comment, LGTM

File: docs/intro/developing.md [hunk 1]
```diff
@@ -64,6 +64,14 @@ You can however inspect the actual and reference pages for web platform reftests
  - `http://localhost:8080/out/webgpu/web_platform/reftests/canvas_clear.https.html`
```
Comment (kainino0x) [inline docs/intro/developing.md:64]: Oh also can update line 60 just above here

File: src/common/tools/run_wpt_ref_tests.ts [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+  await page.addInitScript({
+    content: `
+    (() => {
+      const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+      window.takeScreenshotDelayed = async function(ms) {
+        await wait(ms);
+        document.documentElement.classList.remove('reftest-wait');
+      };
```
[the remark this answers is not in the document]
Comment (greggman) [reply src/common/tools/run_wpt_ref_tests.ts:254]: done

File: docs/intro/developing.md (hunk 1 above)
Comment (greggman) [reply docs/intro/developing.md:64]: done

**Pull request merged**